### PR TITLE
Cs pmml model

### PR DESCRIPTION
**Pull request by CarlosSilva-zd2**
Stable implementation of CollectorService with communication and context detection working as intended for this stage of development. Mechanism in place to change HomeActivity according to detected context and to inform CollectorService of manual context change.
Context is being detected every 10 mins.
Regular ETLs is scheduled for once every week. With checks on the date every 6 hours.
Before trying to run this, please delete data from application. For better results test with GPS turned on.

### CODE REVIEW

File: app/src/main/java/corp/kairos/adamastor/Collector/ActivityRecognitionService.java [hunk 1]
```diff
@@ -38,14 +37,29 @@ protected void onHandleIntent(@Nullable Intent intent) {
...
            if(activity.getConfidence()>mostSignificantValue && activity.getType() != DetectedActivity.ON_FOOT) {
                mostSignificant = activity.getType();
+               mostSignificantValue = activity.getConfidence();
            }
         }
-        Intent result = new  Intent("kairos.CollectorService.ACTIVITY_MONITOR_RESULT");
-        result.putExtra("ActivityNow",mostSignificant);
-        sendBroadcast(result);
+
+        switch(mostSignificant){
```
Comment (Hugainz) [inline app/src/main/java/corp/kairos/adamastor/Collector/ActivityRecognitionService.java]: This switch can be simplified to:
```JAVA
switch(mostSignificant){
    case DetectedActivity.IN_VEHICLE:
    case DetectedActivity.ON_BICYCLE: 
        result = 3;
        break;
    case DetectedActivity.ON_FOOT:
    case DetectedActivity.WALKING:
    case DetectedActivity.RUNNING: 
        result = 2; 
        break;
    case DetectedActivity.STILL:
    case DetectedActivity.UNKNOWN:
    case DetectedActivity.TILTING: 
    default: 
        result = 1;
}
```

File: app/src/main/java/corp/kairos/adamastor/Collector/CollectorService.java [hunk 2]
```diff
@@ -300,11 +474,10 @@ private String getUsername() {
 
     private int getRingtoneMode(){
         int result;
-        AudioManager am = (AudioManager)this.getSystemService(Context.AUDIO_SERVICE);
-        switch(am.getRingerMode()){
-            case AudioManager.RINGER_MODE_NORMAL : result = 0; break;
-            case AudioManager.RINGER_MODE_SILENT : result = 1; break;
-            case AudioManager.RINGER_MODE_VIBRATE : result = 2; break;
+        switch(audioManager.getRingerMode()){
```
Comment (hgg) [inline app/src/main/java/corp/kairos/adamastor/Collector/CollectorService.java:473]: This switch can be simplified as well.

File: app/src/main/java/corp/kairos/adamastor/Collector/ActivityRecognitionService.java (hunk 1 above)
Comment (hgg) [inline app/src/main/java/corp/kairos/adamastor/Collector/ActivityRecognitionService.java:48]: This switch can be simplified to:
```JAVA
switch(mostSignificant){
    case DetectedActivity.IN_VEHICLE:
    case DetectedActivity.ON_BICYCLE: 
        result = 3;
        break;
    case DetectedActivity.ON_FOOT:
    case DetectedActivity.WALKING:
    case DetectedActivity.RUNNING: 
        result = 2; 
        break;
    case DetectedActivity.STILL:
    case DetectedActivity.UNKNOWN:
    case DetectedActivity.TILTING: 
    default: 
        result = 1;
}
```

File: app/src/main/java/corp/kairos/adamastor/Collector/CollectorService.java [hunk 3]
```diff
@@ -235,19 +408,17 @@ private boolean isPhoneActive(){
...
-        AudioManager am = (AudioManager)this.getSystemService(Context.AUDIO_SERVICE);
-        return am.isMusicActive();
+        return audioManager.isMusicActive();
     }
 
     private int getPhoneCallState(){
         int result;
-        TelephonyManager tm = (TelephonyManager)getSystemService(Context.TELEPHONY_SERVICE);
-        switch(tm.getCallState()){
+        switch(telephonyManager.getCallState()){
```
Comment (hgg) [inline app/src/main/java/corp/kairos/adamastor/Collector/CollectorService.java:412]: Switch can be simplified.

File: app/src/main/java/corp/kairos/adamastor/Settings/Settings.java [hunk 4]
```diff
@@ -154,11 +156,60 @@ public UserContext getUserContext(int id) {
...
+
+        this.informCollectorContextChange(result.getContextName());
+
+        return result;
+    }
+
+    private String getContextNamefromCollector(){
+        int context = CollectorService.getInstance().getPredictedContext();
+        String result;
+        switch(context){
```
Comment (hgg) [inline app/src/main/java/corp/kairos/adamastor/Settings/Settings.java:183]: You know what I'm gonna say :joy:

File: app/src/main/java/corp/kairos/adamastor/Settings/Settings.java [hunk 5]
```diff
@@ -154,11 +156,60 @@ public UserContext getUserContext(int id) {
...
+        }
+
+        return result;
+    }
+
+    public void informCollectorContextChange(String contextName){
+
+        int context = 1;
+
+        switch(contextName){
```
Comment (hgg) [inline app/src/main/java/corp/kairos/adamastor/Settings/Settings.java:201]: :rofl: It's my paranoia. I get it if you choose to ignore these remarks :+1:

**Review by hgg — CHANGES_REQUESTED** [review]
This is a huge PR and since it revolves around something I wasn't completely evolved in developing the scope of my comments is very limited. Still, great job :+1: 

I just have a few syntax comments and I'd like you to review the issue with the ModelHandler, not being imported correctly.

**Review by FranciscoRibeiro — APPROVED** [review]
From what I've tested, this looks fine :) We can go ahead and merge

**Review by hgg — DISMISSED** [review]

**Pull request merged**